Introduction.md missing new GitHub workflow

marcoscaceres opened the pull request:
#### df2ff2e1fa1d880f62b8927dc0f1b90aa5b8e333
<pre>
Introduction.md missing new GitHub workflow
<a href="https://bugs.webkit.org/show_bug.cgi?id=242050">https://bugs.webkit.org/show_bug.cgi?id=242050</a>

Adds details for how to use the new GitHub workflow, with `git webkit`.

Reviewed by Jonathan Bedard.

* Introduction.md:

Canonical link: <a href="https://commits.webkit.org/252202@main">https://commits.webkit.org/252202@main</a>
</pre>

youssefsoli commented on Introduction.md:
````diff
@@ -123,12 +107,76 @@ This will allow you to run various tools you by name instead of typing the full
+```Bash
+git add <files-to-add>
+git webkit commit -i <bug number from bugs.webkit.org here>
+```
````
`git-webkit pr` is able to do all these steps at once and might be more user friendly?

marcoscaceres replied:
Tried this out... but didn't quite work if one is already on a custom feature/bug branch.

For example, doing just `git webkit pr` yields commit message:

```
Need a short description (OOPS!).
Need the bug URL (OOPS!).

Reviewed by NOBODY (OOPS!).

* Introduction.md:
```

And doing, for instance, `git webkit pr --issue 242050` errors with: 

```
Creating a pull-request for '242050' but we're on 'introduction'
```

So, I guess the working assumption is that you start fixing stuff on "main" branch, then do `git webkit pr --issue XXXX` to set up the feature branch and generate the commit log. 

(and if you've got you own feature branch, then straight up `git webkit pr` doesn't quite work anymore)

eric-carlson replied:
I also typically work from a local branch and think it would be helpful to include instructions that accommodate this workflow.

youssefsoli commented on Introduction.md:
````diff
@@ -123,12 +107,76 @@ This will allow you to run various tools you by name instead of typing the full
+```Bash
+git commit --amend path/to/new/changes
+git webkit commit -i <bug number from bugs.webkit.org here>
+git webkit pr --overwrite
````
This can be replaced by just doing `git-webkit pr`

JonWBedard commented on Introduction.md:
```diff
@@ -123,12 +107,76 @@ This will allow you to run various tools you by name instead of typing the full
 ### Updating checkouts
 
-There is a script to update a WebKit checkout: `Tools/Scripts/update-webkit`. This script will automatically merge change logs mentioned below.
+There is a script to update a WebKit checkout: `Tools/Scripts/update-webkit`.
```
Probably best to keep this for now, but I do think we should encourage folks to use raw `git` foo for this one.

JonWBedard commented on Introduction.md:
```diff
@@ -123,12 +107,76 @@ This will allow you to run various tools you by name instead of typing the full
+
+That will pull down the details from [bugs.webkit.org](https://bugs.webkit.org) and create a commit message for you. You shouldn't need to make any changes to the auto-generated commit message.
+
+### Submitting a patch
```
This should be "Submitting a Pull-Request"

JonWBedard commented on Introduction.md:
```diff
@@ -263,20 +311,20 @@ then the reason writing a tests is impractical should be explained in the accomp
 It’s in your interest to wait for the Early Warning System to fully build and test your patch on all relevant platforms.
 
-### ChangeLog Files
+### Changelog
```
This should be `Commit Messages`

JonWBedard commented on Introduction.md:
````diff
@@ -263,20 +311,20 @@ then the reason writing a tests is impractical should be explained in the accomp
+
+```Bash
+git webkit commit -i <your bug number here>
+```
````
This isn't quite right. Commit messages are generated by a `git` hook (`.git/hooks/prepare-commit-msg`, specifically) that is configured during `git-webkit setup` and derived from `Tools/Scripts/hooks/prepare-commit-msg`. I'm not sure how much of that we want to say, though. Maybe just "`git-webkit setup` configures your `git` hooks to properly configure your commit message"?

marcoscaceres replied:
Ah, makes sense. Thanks for clarifying.

JonWBedard commented on Introduction.md:
````diff
@@ -129,6 +113,50 @@ There is a script to update a WebKit checkout: `Tools/Scripts/update-webkit`. Th
+```
+
+That will pull down the details from [bugs.webkit.org](https://bugs.webkit.org), create a new git branch, and generate a commit message for you.
+You shouldn't need to make any changes to the auto-generated commit message.
````
This isn't true, right? We expect contributors to add descriptions of what they are changing into their commit message.

marcoscaceres replied:
Ah, I've never done that for any patches as I was scared of breaking the auto-generated commit message... or that it would get trashed after making changes. The old changelogs files were a bit temperamental, so I always trashed them and regenerated them.   

Let me try it by fixing this sentence...

JonWBedard commented on Introduction.md:
````diff
@@ -129,6 +113,50 @@ There is a script to update a WebKit checkout: `Tools/Scripts/update-webkit`. Th
+git webkit pr
+```
+
+That will replace your old pull request with a new one with the new changes, while also updating the commit message.
````
I think you mean "updating the PR description with your current commit message"

marcoscaceres committed:
Introduction.md missing new GitHub workflow
https://bugs.webkit.org/show_bug.cgi?id=242050

Adds details for how to use the new GitHub workflow, with `git webkit`.

Reviewed by Jonathan Bedard.

* Introduction.md:

Canonical link: https://commits.webkit.org/252202@main
